src/ftxui-grid-container/grid-container.cpp

```diff
@@ -0,0 +1,45 @@
+#include <utility>
+#include <vector>
+
+#include <ftxui/component/component.hpp>
+#include <ftxui/component/component_base.hpp>
+#include <ftxui/dom/elements.hpp>
+
+#include <ftxui-grid-container/grid-container.hpp>
+
+namespace ftxui {
+namespace {
+class Impl : public ComponentBase {
+ private:
+  std::vector<Components> rows_;
+  Ref<int> selector_vertical_;
+  Ref<int> selector_horizontal_;
+
+ public:
+  //        NOLINTNEXTLINE
+  Impl(std::vector<Components> lines,
+       Ref<int> selector_vertical,
+       Ref<int> selector_horizontal)
+      : rows_(std::move(lines)),
+        selector_vertical_(selector_vertical),
+        selector_horizontal_(selector_horizontal) {
+    Components horizontalComponents_;
```

Comment:
@ArthurSonzogni newbie question... why won't horizontalCompoents_ go out of scope after the constructor ends?

Reply:
It will go out of scope.

However, `Container::Vertical` is going to unwrap every individual Component and add them as its children. They are now owned by the `Container::Vertical`.

Reply:
Ok thanks!